daml-lf/spec/daml-lf-1.rst

```diff
@@ -565,12 +560,24 @@ strings as *package identifiers*.  ::
   Package versions
            pversion ::= PackageVersionString        -- PackageVersion
 
-We do not specify an explicit syntax for contract identifiers as it is
-not possible to refer to them statically within a program. In
-practice, contract identifiers can be created dynamically through
-interactions with the underlying ledger. See the `operation semantics
-of update statements <Update Interpretation_>`_ for the formal
-specification of those interactions.
+  V0 Contract identifiers:
+          cidV0  ∈  #[a-zA-Z0-9\._:-#/ ]+           -- V0ContractId
+
+  V1 Contract identifiers:
+          cidV1  ∈  00([0-9a-f][0-9a-f]){32,190}    -- V1ContractId
```

Comment:
Why is 190 the upper limit? I don’t have an issue with it. Just wondering where the magic number comes from.

Reply:
You are so quick ;) 
I change it to 96 (3 x 32 bytes)

Reply:
:+1: